File: Old Version/ReportApp/ReportApp/Data/Report.swift
//
//  Report.swift
//  ReportApp
//
//  Created by 葉家均 on 2022/12/8.
//

import Foundation

struct ReportContent: Decodable, Identifiable, Equatable {
    var id: Int
    var dateTime: String
    var reportDetail: [ReportDetail]
    var imageDetail: [ImageDetail]
    var yearMonth:String?
    var day: String?
    var dateValue: Date?
}

struct ReportDetail: Decodable, Equatable,Identifiable {
    var id: Int
    var content: String
}

struct ImageDetail: Decodable, Equatable,Identifiable {
    var id: Int
    var imgPath: String
}

File: report_app/lib/data/report.dart
import 'package:report_app/data/image_detail.dart';
import 'package:report_app/data/report_detail.dart';

class Report {
  int id = 0;
  String dateTime = "";
  List<ReportDetail> reportDetail = [];
  List<ImageDetail> imageDetail = [];

  Map<String, dynamic> toJson() => {
    'id': id,
    "dateTime": dateTime,
    "reportDetail": reportDetail,
    "imageDetail": imageDetail
  };

  Report.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        dateTime = json["dateTime"],
        reportDetail = (json["reportDetail"] as List).map((e) => ReportDetail.fromJson(e)).toList(),
        imageDetail = (json["imageDetail"] as List).map((e) => ImageDetail.fromJson(e)).toList();
}